src/applications/upgradeable_contract/upgradeable_contract.md

```diff
@@ -0,0 +1,14 @@
+# Upgradeable Contract
+
+Starknet distinguishes between a contract and its implementation by separating contracts into classes and instances.
+
+A contract class is the definition of the contract: Cairo byte code, hint information, entry point names, and everything that defines its semantics unambiguously.
+
+Each class is identified by its class hash, which is analogous to a class name in an object-oriented programming language. A contract instance is a deployed contract corresponding to a class.
```

Comment:
As I suggested to add the POO comparison right at the start, maybe remove/change it there:
"To identify different contract classes, Starknet assigns a unique identifier to each class: the **class hash**. A contract instance is a deployed contract that corresponds to a specific contract class. Think of it as an instance of an object in languages like Java or Python."